cfme/tests/openstack/infrastructure/test_roles.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import re
+from cfme.fixtures import pytest_selenium as sel
+from cfme.infrastructure.cluster import Cluster
+from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from cfme.web_ui import InfoBlock, Quadicon
+from utils import testgen
+from utils.appliance.implementations.ui import navigate_to
+from utils.version import current_version
+
+
+pytest_generate_tests = testgen.generate(testgen.providers_by_class,
+                                         [OpenstackInfraProvider],
+                                         scope='module')
+pytestmark = [pytest.mark.uncollectif(lambda: current_version() < '5.7'),
+              pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+ROLES = ['NovaCompute', 'Controller', 'Compute', 'BlockStorage', 'SwiftStorage',
+         'CephStorage']
+
+
+def test_host_role_type(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    for quad in my_quads:
+        role_name = str(quad.name)
+        role_name = re.search(r'\((\w+)\)', role_name).group(1)
```

Comment:
That's fine, it can stay as-is.

To use translate here, you would use None as the first argument - the 'translation table' - which will pass the non-deleted characters through unchanged.

```python
string = '(my_role_name)'
print string.translate(None, '()')
# output: my_role_name
```